### PR TITLE
Extended artifact test

### PR DESCRIPTION
**Pull request by petemoore**
Added some checks to the artifact test, and specify charset in mimetype now.

### CODE REVIEW

File: plugins/livelog/livelog.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func (tp *taskPlugin) Prepare(context *runtime.TaskContext) error {
 
 	err := tp.context.CreateRedirectArtifact(runtime.RedirectArtifact{
 		Name:     "public/logs/live.log",
-		Mimetype: "text/plain",
+		Mimetype: "text/plain; charset=utf-8",
```
Comment (jonasfj) [inline plugins/livelog/livelog.go]: Probably best to make this a constant...
`var mimePlainText = "text/plain; charset=utf-8"`

And then reuse that everywhere... Just so that we only have it once this package...

File: test/helper_non-windows_test.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// +build !windows
```
Comment (jonasfj) [inline test/helper_non-windows_test.go]: file name should be `_unix_test.go` check what pattern we use in `engine/native/system/` package.

Comment (petemoore) [inline test/helper_non-windows_test.go]: bsd/darwin is not unix - so I suggest the other naming should be changed, not this. !windows _means_ non-windows - they are equivalent. unix does not mean non-windows - those are not the only two variants.

File: plugins/livelog/livelog.go (hunk 1 above)
Comment (petemoore) [inline plugins/livelog/livelog.go]: yeah maybe - let's do in a different PR

File: test/helper_non-windows_test.go (hunk 2 above)
Comment (jonasfj) [inline test/helper_non-windows_test.go]: > bsd/darwin is not unix

@petemoore, afaik the BSDs including OS X actually derive from the original unix :)
OS X has certainly been unix certified at times... Regardless, they are all consider unix-like, just like linux (which also isn't unix).
We could also call it `_posix`, but I don't enjoy dashes in golang filenames, it just feels wrong...
